### PR TITLE
[Reactant] Try building Reactant with a more up-to-date clang

### PR DESCRIPTION
**Pull request by mofeing**
There is a very weird bug in Reactant which only appears on Yggdrasil-built aarch64 macOS artifact. If we build it with LLVM 19 on local, it works fine.

We are trying to check if it's due to the LLVM version used for compilation.

Do not merge yet please.

### CODE REVIEW

File: R/Reactant/build_tarballs.jl [hunk 1]
```diff
@@ -6,10 +6,10 @@ include(joinpath(YGGDRASIL_DIR, "fancy_toys.jl"))
 
 name = "Reactant"
 repo = "https://github.com/EnzymeAD/Reactant.jl.git"
-version = v"0.0.19"
+version = v"0.0.19+1"
```
Comment (mofeing) [inline R/Reactant/build_tarballs.jl:9]:
```suggestion
version = v"0.0.19"
```

**Comment by wsmoses** [comment]
```
[23:23:11] ERROR: /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/external/zlib/BUILD.bazel:5:11: Compiling zutil.c failed: (Exit 1): x86_64-apple-darwin14-cc failed: error executing command (from target @zlib//:zlib)
--
  | [23:23:11]   (cd /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/sandbox/processwrapper-sandbox/1879/execroot/__main__ && \
  | [23:23:11]   exec env - \
  | [23:23:11]     JULIA='' \
  | [23:23:11]     LD_LIBRARY_PATH=/usr/lib/csl-musl-x86_64:/usr/local/lib64:/usr/local/lib:/usr/lib64:/usr/lib:/lib64:/lib:/workspace/x86_64-linux-musl-cxx11/destdir/lib:/workspace/x86_64-linux-musl-cxx11/destdir/lib64:/opt/x86_64-linux-musl/x86_64-linux-musl/lib64:/opt/x86_64-linux-musl/x86_64-linux-musl/lib:/opt/x86_64-apple-darwin14/x86_64-apple-darwin14/lib:/opt/x86_64-apple-darwin14/lib:/workspace/destdir/lib64:/workspace/destdir/lib \
  | [23:23:11]     MACOSX_DEPLOYMENT_TARGET=10.14 \
  | [23:23:11]     PATH=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.local/bin:/opt/bin/x86_64-apple-darwin14-libgfortran5-cxx11-gpu+none-mode+opt:/opt/x86_64-apple-darwin14/bin:/opt/bin/x86_64-linux-musl-cxx11:/opt/x86_64-linux-musl/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:/workspace/x86_64-linux-musl-cxx11/destdir/bin:/workspace/destdir/bin \
  | [23:23:11]     PWD=/proc/self/cwd \
  | [23:23:11]     TEMP=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.tmp \
  | [23:23:11]     TMP=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.tmp \
  | [23:23:11]     TMPDIR=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.tmp \
  | [23:23:11]   /opt/bin/x86_64-apple-darwin14-libgfortran5-cxx11-gpu+none-mode+opt/x86_64-apple-darwin14-cc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections -MD -MF bazel-out/k8-opt/bin/external/zlib/_objs/zlib/zutil.pic.d '-frandom-seed=bazel-out/k8-opt/bin/external/zlib/_objs/zlib/zutil.pic.o' -fPIC '-DBAZEL_CURRENT_REPOSITORY="zlib"' -iquote external/zlib -iquote bazel-out/k8-opt/bin/external/zlib -isystem external/zlib -isystem bazel-out/k8-opt/bin/external/zlib -Wno-shift-negative-value -DZ_HAVE_UNISTD_H -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/zlib/zutil.c -o bazel-out/k8-opt/bin/external/zlib/_objs/zlib/zutil.pic.o)
  | [23:23:11] # Configuration: 527a6f5445d452177076fb79ac473953eeec59bee5e0dcc4f295b128b5a4fe60
  | [23:23:11] # Execution platform: @local_execution_config_platform//:platform
  | [23:23:11]
  | [23:23:11] Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
  | [23:23:11] In file included from external/zlib/zutil.c:8:
  | [23:23:11] external/zlib/zutil.h:170:11: warning: 'OS_CODE' macro redefined [-Wmacro-redefined]
  | [23:23:11]   170 \| #  define OS_CODE 19
  | [23:23:11]       \|           ^
  | [23:23:11] external/zlib/zutil.h:141:11: note: previous definition is here
  | [23:23:11]   141 \| #  define OS_CODE  7
  | [23:23:11]       \|           ^
  | [23:23:11] In file included from external/zlib/zutil.c:10:
  | [23:23:11] In file included from external/zlib/gzguts.h:21:
  | [23:23:11] In file included from /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/c++/v1/stdio.h:108:
  | [23:23:11] /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/stdio.h:214:7: error: expected identifier or '('
  | [23:23:11]   214 \| FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));
  | [23:23:11]       \|          ^
  | [23:23:11] external/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
  | [23:23:11]   147 \| #        define fdopen(fd,mode) NULL /* No fdopen() */
  | [23:23:11]       \|                                 ^
  | [23:23:11] /opt/x86_64-linux-musl/lib/clang/18/include/__stddef_null.h:26:16: note: expanded from macro 'NULL'
  | [23:23:11]    26 \| #define NULL ((void*)0)
  | [23:23:11]       \|                ^
  | [23:23:11] In file included from external/zlib/zutil.c:10:
  | [23:23:11] In file included from external/zlib/gzguts.h:21:
  | [23:23:11] In file included from /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/c++/v1/stdio.h:108:
  | [23:23:11] /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/stdio.h:214:7: error: expected ')'
  | [23:23:11] external/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
  | [23:23:11]   147 \| #        define fdopen(fd,mode) NULL /* No fdopen() */
  | [23:23:11]       \|                                 ^
  | [23:23:11] /opt/x86_64-linux-musl/lib/clang/18/include/__stddef_null.h:26:16: note: expanded from macro 'NULL'
  | [23:23:11]    26 \| #define NULL ((void*)0)
  | [23:23:11]       \|                ^
  | [23:23:11] /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/stdio.h:214:7: note: to match this '('
  | [23:23:11] external/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
  | [23:23:11]   147 \| #        define fdopen(fd,mode) NULL /* No fdopen() */
  | [23:23:11]       \|                                 ^
  | [23:23:11] /opt/x86_64-linux-musl/lib/clang/18/include/__stddef_null.h:26:15: note: expanded from macro 'NULL'
  | [23:23:11]    26 \| #define NULL ((void*)0)
  | [23:23:11]       \|               ^
  | [23:23:11] In file included from external/zlib/zutil.c:10:
  | [23:23:11] In file included from external/zlib/gzguts.h:21:
  | [23:23:11] In file included from /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/c++/v1/stdio.h:108:
  | [23:23:11] /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/sys-root/usr/include/stdio.h:214:7: error: expected ')'
  | [23:23:11]   214 \| FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));
  | [23:23:11]       \|          ^
  | [23:23:11] external/zlib/zutil.h:147:


```

**Comment by mofeing** [comment]
Similar issue https://github.com/pocoproject/poco/issues/4586

**Comment by wsmoses** [comment]
```
sandbox:${WORKSPACE}/srcdir/Reactant.jl/deps/ReactantExtra # /opt/bin/x86_64-linux-musl-cxx11/x86_64-linux-musl-clang -U_FORTIFY_SOURCE -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections -fdata-sections '-stdlib=libstdc++' -MD -MF bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/Support/Threading.d '-frandom-seed=bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/Support/Threading.o' '-DLLVM_ON_UNIX=1' '-DNO_HAVE_BACKTRACE=0' '-DBACKTRACE_HEA
DER=<execinfo.h>' '-DLTDL_SHLIB_EXT=".so"' '-DLLVM_PLUGIN_EXT=".so"' '-DLLVM_ENABLE_THREADS=0' '-DHAVE_DEREGISTER_FRAME=1' '-DHAVE_LIBPTHREAD=1' '-DHAVE_PTHREAD_GETNAME_NP=0' '-DHAVE_PTHREAD_H=1' '-DHAVE_PTHREAD_SETNAME_NP=0' '-DHAVE_REGISTER_FR
AME=1' '-DHAVE_SETENV_R=1' '-DHAVE_STRERROR_R=1' '-DHAVE_SYSEXITS_H=1' '-DHAVE_UNISTD_H=1' -D_GNU_SOURCE '-DDONT_HAVE_ANY_MALLINFO=0' '-DHAVE_SBRK=1' '-DHAVE_STRUCT_STAT_ST_MTIM_TV_NSEC=1' -DHAVE_BUILTIN_THREAD_POINTER '-DLLVM_NATIVE_ARCH="X86"' '-DLLVM_NATIVE_ASMPARSER=LLVMInitializeX86AsmParser' '-DLLVM_NATIVE_ASMPRINTER=LLVMInitializeX86AsmPrinter' '-DLLVM_NATIVE_DISASSEMBLER=LLVMInitializeX86Disassembler' '-DLLVM_NATIVE_TARGET=LLVMInitializeX86Target' '-DLLVM_NATIVE_TARGETINFO=LLVMInitializeX86TargetInfo' '-DLLVM_NATIVE_TARGETMC=LLVMInitializeX86TargetMC' '-DLLVM_NATIVE_TARGETMCA=LLVMInitializeX86TargetMCA' '-DLLVM_HOST_TRIPLE="x86_64-unknown-linux-gnu"' '-DLLVM_DEFAULT_TARGET_TRIPLE="x86_64-unknown-linux-gnu"' '-DLLVM_VE
RSION_MAJOR=20' '-DLLVM_VERSION_MINOR=0' '-DLLVM_VERSION_PATCH=0' '-DLLVM_VERSION_STRING="20.0.0git"' -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS '-DLLVM_HAS_AArch64_TARGET=1' '-DLLVM_HAS_AMDGPU_TARGET=1' '-DLLVM_HAS_ARM_TARGET=1' '-DLLVM_HAS_NVPTX_TARGET=1' '-DLLVM_HAS_PowerPC_TARGET=1' '-DLLVM_HAS_RISCV_TARGET=1' '-DLLVM_HAS_SystemZ_TARGET=1' '-DLLVM_HAS_X86_TARGET=1' '-DBLAKE3_USE_NEON=0' -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512 -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 '-DBAZEL_CURRENT_REPOSITORY="llvm-project"' -iquote external/llvm-project -iquote bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project -isystem external/llvm-project/llvm/include -isystem bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/include -g0 -g0 '-std=c++17' -w '-std=c++17' '--sysroot=/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root' -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"
' -Wno-unused-command-line-argument -Wno-gnu-offsetof-extensions -c external/llvm-project/llvm/lib/Support/Threading.cpp -o bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/Support/Threading.o -v
clang version 16.0.6 (/home/gbaraldi/.julia/dev/BinaryBuilderBase/deps/downloads/clones/llvm-project.git-1df819a03ecf6890e3787b27bfd4f160aeeeeacd50a98d003be8b0893f11a9be 7cbf1a2591520c2491aa35339f227775f4d3adf6)
Target: x86_64-unknown-linux-musl
Thread model: posix
InstalledDir: /opt/x86_64-linux-musl/bin
Selected GCC installation: /opt/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/10.2.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
 (in-process)
 "/opt/x86_64-linux-musl/bin/clang" -cc1 -triple x86_64-unknown-linux-musl -emit-obj -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name Threading.cpp -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=none -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -ffunction-sections -fdata-sections -fcoverage-compilation-dir=/workspace/srcdir/Reactant.jl/deps/ReactantExtra -resource-dir /opt/x86_64-linux-musl/lib/clang/16 -dependency-file bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/Support/Threading.d -MT bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/Support/Threading.o -sys-header-deps -isystem /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0 -isystem /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0/x86_64-linux-musl -isystem /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0/backward -isystem /opt/x86_64-linux-musl/x86_64-linux-musl/include -isystem /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/include -iquote external/llvm-project -iquote bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project -isystem external/llvm-project/llvm/include -isystem bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/include -D _GLIBCXX_USE_CXX11_ABI=1 -U _FORTIFY_SOURCE -D _FORTIFY_SOURCE=1 -D NDEBUG -D LLVM_ON_UNIX=1 -D NO_HAVE_BACKTRACE=0 -D BACKTRACE_HEADER=<execinfo.h> -D "LTDL_SHLIB_EXT=\".so\"" -D "LLVM_PLUGIN_EXT=\".so\"" -D LLVM_ENABLE_THREADS=0 -D HAVE_DEREGISTER_FRAME=1 -D HAVE_LIBPTHREAD=1 -D HAVE_PTHREAD_GETNAME_NP=0 -D HAVE_PTHREAD_H=1 -D HAVE_PTHREAD_SETNAME_NP=0 -D HAVE_REGISTER_FRAME=1 -D HAVE_SETENV_R=1 -D HAVE_STRERROR_R=1 -D HAVE_SYSEXITS_H=1 -D HAVE_UNISTD_H=1 -D _GNU_SOURCE -D DONT_HAVE_ANY_MALLINFO=0 -D HAVE_SBRK=1 -D HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC=1 -D HAVE_BUILTIN_THREAD_POINTER -D "LLVM_NATIVE_ARCH=\"X86\"" -D LLVM_NATIVE_ASMPARSER=LLVMInitializeX86AsmParser -D LLVM_NATIVE_ASMPRINTER=LLVMInitializeX86AsmPrinter -D LLVM_NATIVE_DISASSEMBLER=LLVMInitializeX86Disassembler -D LLVM_NATIVE_TARGET=LLVMInitializeX86Target -D LLVM_NATIVE_TARGETINFO=LLVMInitializeX86TargetInfo -D LLVM_NATIVE_TARGETMC=LLVMInitializeX86TargetMC -D LLVM_NATIVE_TARGETMCA=LLVMInitializeX86TargetMCA -D "LLVM_HOST_TRIPLE=\"x86_64-unknown-linux-gnu\"" -D "LLVM_DEFAULT_TARGET_TRIPLE=\"x86_64-unknown-linux-gnu\"" -D LLVM_VERSION_MAJOR=20 -D LLVM_VERSION_MINOR=0 -D LLVM_VERSION_PATCH=0 -D "LLVM_VERSION_STRING=\"20.0.0git\"" -D __STDC_LIMIT_MACROS -D __STDC_CONSTANT_MACROS -D __STDC_FORMAT_MACROS -D LLVM_HAS_AArch64_TARGET=1 -D LLVM_HAS_AMDGPU_TARGET=1 -D LLVM_HAS_ARM_TARGET=1 -D LLVM_HAS_NVPTX_TARGET=1 -D LLVM_HAS_PowerPC_TARGET=1 -D LLVM_HAS_RISCV_TARGET=1 -D LLVM_HAS_SystemZ_TARGET=1 -D LLVM_HAS_X86_TARGET=1 -D BLAKE3_USE_NEON=0 -D BLAKE3_NO_AVX2 -D BLAKE3_NO_AVX512 -D BLAKE3_NO_SSE2 -D BLAKE3_NO_SSE41 -D "BAZEL_CURRENT_REPOSITORY=\"llvm-project\"" -D "__DATE__=\"redacted\"" -D "__TIMESTAMP__=\"redacted\"" -D "__TIME__=\"redacted\"" -isysroot /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root -internal-isystem /opt/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/10.2.0/../../../../x86_64-linux-musl/include/c++/10.2.0 -internal-isystem /opt/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/10.2.0/../../../../x86_64-linux-musl/include/c++/10.2.0/x86_64-linux-musl -internal-isystem /opt/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/10.2.0/../../../../x86_64-linux-musl/include/c++/10.2.0/backward -internal-isystem /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/local/include -internal-isystem /opt/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/10.2.0/../../../../x86_64-linux-musl/include -internal-externc-isystem /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/include -internal-externc-isystem /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include -internal-isystem /opt/x86_64-linux-musl/lib/clang/16/include -source-date-epoch 0 -O2 -Wno-builtin-macro-redefined -Wno-unused-command-line-argument -Wno-gnu-offsetof-extensions -w -std=c++17 -fdeprecated-macro -fdebug-compilation-dir=/workspace/srcdir/Reactant.jl/deps/ReactantExtra -ferror-limit 19 -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -fcolor-diagnostics -vectorize-loops -vectorize-slp -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/_objs/Support/Threading.o -x c++ external/llvm-project/llvm/lib/Support/Threading.cpp
clang -cc1 version 16.0.6 based upon LLVM 16.0.6 default target x86_64-unknown-linux-musl
ignoring nonexistent directory "/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/include"
ignoring nonexistent directory "/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/local/include"
ignoring nonexistent directory "/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/include"
ignoring duplicate directory "/opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0"
ignoring duplicate directory "/opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0/x86_64-linux-musl"
ignoring duplicate directory "/opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0/backward"
ignoring duplicate directory "/opt/x86_64-linux-musl/x86_64-linux-musl/include"
#include "..." search starts here:
 external/llvm-project
 bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project
#include <...> search starts here:
 /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0
 /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0/x86_64-linux-musl
 /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/10.2.0/backward
 /opt/x86_64-linux-musl/x86_64-linux-musl/include
 external/llvm-project/llvm/include
 bazel-out/k8-opt-exec-50AE0418/bin/external/llvm-project/llvm/include
 /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include
 /opt/x86_64-linux-musl/lib/clang/16/include
End of search list.
In file included from external/llvm-project/llvm/lib/Support/Threading.cpp:69:
external/llvm-project/llvm/lib/Support/Unix/Threading.inc:250:5: error: no member named 'pthread_getname_np' in the global namespace
  ::pthread_getname_np(::pthread_self(), buf, len);
  ~~^
1 error generated.
```

**Comment by wsmoses** [comment]
https://github.com/llvm/llvm-project/pull/106486

**Comment by wsmoses** [comment]
https://github.com/llvm/llvm-project/pull/110854

**Comment by wsmoses** [comment]
@mofeing okay macos error is now:
```

[15:22:04] ERROR: /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/external/xla/xla/service/cpu/BUILD:1190:11: Compiling xla/service/cpu/runtime_matmul_acl.cc failed: (Exit 1): aarch64-apple-darwin20-cc failed: error executing command (from target @xla//xla/service/cpu:runtime_matmul_acl)
--
  | [15:22:04]   (cd /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/sandbox/processwrapper-sandbox/1724/execroot/__main__ && \
  | [15:22:04]   exec env - \
  | [15:22:04]     JULIA='' \
  | [15:22:04]     LD_LIBRARY_PATH=/usr/lib/csl-musl-x86_64:/usr/local/lib64:/usr/local/lib:/usr/lib64:/usr/lib:/lib64:/lib:/workspace/x86_64-linux-musl-cxx11/destdir/lib:/workspace/x86_64-linux-musl-cxx11/destdir/lib64:/opt/x86_64-linux-musl/x86_64-linux-musl/lib64:/opt/x86_64-linux-musl/x86_64-linux-musl/lib:/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/lib:/opt/aarch64-apple-darwin20/lib:/workspace/destdir/lib64:/workspace/destdir/lib \
  | [15:22:04]     MACOSX_DEPLOYMENT_TARGET=10.14 \
  | [15:22:04]     PATH=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.local/bin:/opt/bin/aarch64-apple-darwin20-libgfortran5-cxx11-gpu+none-mode+dbg:/opt/aarch64-apple-darwin20/bin:/opt/bin/x86_64-linux-musl-cxx11:/opt/x86_64-linux-musl/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:/workspace/x86_64-linux-musl-cxx11/destdir/bin:/workspace/destdir/bin \
  | [15:22:04]     PWD=/proc/self/cwd \
  | [15:22:04]     TEMP=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.tmp \
  | [15:22:04]     TMP=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.tmp \
  | [15:22:04]     TMPDIR=/workspace/srcdir/Reactant.jl/deps/ReactantExtra/.tmp \
  | [15:22:04]   /opt/bin/aarch64-apple-darwin20-libgfortran5-cxx11-gpu+none-mode+dbg/aarch64-apple-darwin20-cc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g '-std=c++17' -MD -MF bazel-out/k8-dbg/bin/external/xla/xla/service/cpu/_objs/runtime_matmul_acl/runtime_matmul_acl.pic.d '-frandom-seed=bazel-out/k8-dbg/bin/external/xla/xla/service/cpu/_objs/runtime_matmul_acl/runtime_matmul_acl.pic.o' -fPIC '-DEIGEN_MAX_ALIGN_BYTES=64' -DEIGEN_ALLOW_UNALIGNED_SCALARS '-DEIGEN_USE_AVX512_GEMM_KERNELS=0' '-DBAZEL_CURRENT_REPOSITORY="xla"' -iquote external/xla -iquote bazel-out/k8-dbg/bin/external/xla -iquote external/com_google_absl -iquote bazel-out/k8-dbg/bin/external/com_google_absl -iquote external/eigen_archive -iquote bazel-out/k8-dbg/bin/external/eigen_archive -iquote external/tsl -iquote bazel-out/k8-dbg/bin/external/tsl -iquote external/nsync -iquote bazel-out/k8-dbg/bin/external/nsync -iquote external/ml_dtypes -iquote bazel-out/k8-dbg/bin/external/ml_dtypes -Ibazel-out/k8-dbg/bin/external/ml_dtypes/_virtual_includes/float8 -Ibazel-out/k8-dbg/bin/external/ml_dtypes/_virtual_includes/intn -isystem external/eigen_archive -isystem bazel-out/k8-dbg/bin/external/eigen_archive -isystem external/eigen_archive/mkl_include -isystem bazel-out/k8-dbg/bin/external/eigen_archive/mkl_include -isystem external/nsync/public -isystem bazel-out/k8-dbg/bin/external/nsync/public -isystem external/ml_dtypes -isystem bazel-out/k8-dbg/bin/external/ml_dtypes -isystem external/ml_dtypes/ml_dtypes -isystem bazel-out/k8-dbg/bin/external/ml_dtypes/ml_dtypes '-D__ARM_FEATURE_AES=1' '-D__ARM_NEON=1' '-D__ARM_FEATURE_SHA2=1' '-std=c++17' -w '-std=c++17' '-DTCP_USER_TIMEOUT=0' -DEIGEN_AVOID_STL_ARRAY -Iexternal/gemmlowp -Wno-sign-compare '-ftemplate-depth=900' -fno-exceptions -msse3 -pthread -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/xla/xla/service/cpu/runtime_matmul_acl.cc -o bazel-out/k8-dbg/bin/external/xla/xla/service/cpu/_objs/runtime_matmul_acl/runtime_matmul_acl.pic.o)
  | [15:22:04] # Configuration: bc40b856360704881882828c58d2726b4017a3525f9761f7e5a99d0e5c9cc043
  | [15:22:04] # Execution platform: @local_execution_config_platform//:platform
  | [15:22:04]
  | [15:22:04] Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbo


```

**Comment by wsmoses** [comment]
coming from presumably https://github.com/openxla/xla/blob/849f758a85a713e7f5fbe0de0c3df363935aac4f/xla/tsl/tsl.bzl#L315

**Comment by mofeing** [comment]
Ups, now `x86_64-linux-gnu` with `opt` mode crashes due to rocm:

```
[16:49:43] [0 / 6] [Prepa] BazelWorkspaceStatusAction stable-status.txt
[16:49:43] ERROR: /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/external/local_config_rocm/rocm/BUILD:29:11: SolibSymlink _solib_local/_U@local_Uconfig_Urocm_S_Srocm_Chip___Urocm_Srocm_Slib/libhip.so failed: missing input file '@local_config_rocm//rocm:rocm/lib/libhip.so'
[16:49:43] ERROR: /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/external/local_config_rocm/rocm/BUILD:29:11: SolibSymlink _solib_local/_U@local_Uconfig_Urocm_S_Srocm_Chip___Urocm_Srocm_Slib/libhip.so failed: 1 input file(s) do not exist
[16:49:43] Target //:libReactantExtra.so failed to build
[16:49:43] ERROR: /workspace/bazel_root/097636303b1142f44508c1d8e3494e4b/external/local_config_rocm/rocm/BUILD:29:11 SolibSymlink _solib_local/_U@local_Uconfig_Urocm_S_Srocm_Chip___Urocm_Srocm_Slib/libhip.so failed: 1 input file(s) do not exist
```

**Comment by wsmoses** [comment]
yeah I'm trying to figure that one out in bg, @mofeing can you try to figure out the macos side?

**Comment by mofeing** [comment]
I can try, but Bazel beats me every time

**Comment by mofeing** [comment]
Current error in `aarch64-apple-darwin` is
```
In file included from external/onednn/src/common/gemm.cpp:27:
In file included from external/onednn/src/cpu/gemm/gemm.hpp:47:
external/onednn/src/cpu/aarch64/cpu_isa_traits.hpp:35:10: fatal error: 'cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h' file not found
   35 | #include "cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

but the flag `-isystem external/onednn/src` is set, where the header should be located https://github.com/oneapi-src/oneDNN/blob/main/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h

**Comment by mofeing** [comment]
About the linker issue, the problem might be originating in https://github.com/openxla/xla/blob/29cad9d7435d99fd85ca1c2c3a32fd90041e9b10/tools/toolchains/cross_compile/cc/cc_toolchain_config.bzl#L841-L843 or in https://github.com/openxla/xla/blob/29cad9d7435d99fd85ca1c2c3a32fd90041e9b10/tools/toolchains/cpus/aarch64/crosstool/cc_toolchain_config.bzl.tpl#L169

**Pull request merged**